### PR TITLE
CompileSync

### PR DESCRIPTION
**Pull request by yormi**
Here it is, as mentionned here: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/elm-dev/gAyoEsQtHrU

Would be nice once merge so that I can fix the dependency there: https://github.com/yormi/babel-plugin-elm

The test `#compileWorker works with BasicWorker.elm` is failing as it was before this addition. I saw someone just opened an issue about it.

Let me know if there is anything !

### CODE REVIEW

**Comment by rtfeldman** [comment]
@yormi sorry I haven't had a chance to review this yet...it's on my radar though!

**Comment by yormi** [comment]
I guess your pretty busy. I'll be patient. I can use the plugin with the fork anyway :)

File: test/fixtures/elm-package.json [hunk 1]
```diff
@@ -9,8 +9,8 @@
     "native-modules": true,
     "exposed-modules": [],
     "dependencies": {
-        "elm-lang/core": "4.0.0 <= v < 5.0.0",
-        "elm-lang/html": "1.0.0 <= v < 2.0.0"
+        "elm-lang/core": "5.0.0 <= v < 6.0.0",
+        "elm-lang/html": "2.0.0 <= v < 3.0.0"
     },
-    "elm-version": "0.17.0 <= v < 0.18.0"
+    "elm-version": "0.18.0 <= v < 0.19.0"
```
Comment (rtfeldman) [inline test/fixtures/elm-package.json]: Ha, nice!

File: test/compileSync.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+var chai = require("chai")
+var spies = require("chai-spies");
+var path = require("path");
+var compiler = require(path.join(__dirname, ".."));
+
+chai.use(spies);
```
Comment (yormi) [inline test/compileSync.js]: Not sure what you are talking about... The requires are all used in the tests. And if you are talking about the tests themselves, them made me find bugs. When I was calling the wrong `spawnFn` for instance. 

Does that answer the question or I missed the point !? Let me know :)

Comment (rtfeldman) [inline test/compileSync.js]: Sorry, I meant specifically `chai.use(spies)`

Comment (yormi) [inline test/compileSync.js]: Ahhhh ! You are right !
I thought I needed that to use the assertion on the `emitWarning` spy.
Removing !

**Comment by rtfeldman** [comment]
Looks like some CI failures are due to https://github.com/rtfeldman/node-elm-compiler/blob/master/.travis.yml#L14 - and https://github.com/rtfeldman/node-elm-compiler/blob/master/appveyor.yml#L2 - both need to be changed to `0.18.0`

**Comment by yormi** [comment]
Only the failing basic worker test left. It was already failing I think. I don't know where the problem come from. Any ideas ?

**Comment by rtfeldman** [comment]
Hm...it's passing on `master` - maybe try rebasing on upstream `master`?

**Comment by yormi** [comment]
Ok I rebased on the upstream main and it still fails the `BasicWorker` test. 
```
`I cannot find module 'Html.App'.`

I don't have time today to check further but so far, I found that the problem arises with the version bump to 0.18.

**Comment by yormi** [comment]
Ahhh voilà ! Got my answer [here](https://groups.google.com/forum/#!msg/elm-discuss/FdmuPgEXXBo/Es2jaKb6BAAJ) ;)

**Comment by rtfeldman** [comment]
Aha, lovely! Thanks @yormi!

**Comment by rtfeldman** [comment]
Published as `4.3.0`.

**Pull request merged**